src/App.tsx

```diff
@@ -1,113 +1,91 @@
-      .finally(() => setLoaded(true));
-  }
+  useEffect(() => {
+    dispatch(usersActions.initUsers());
+  }, []);
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(postsActions.selectPost(null));
 
-    if (author) {
-      loadUserPosts(author.id);
-    } else {
-      setPosts([]);
+    if (useR) {
+      dispatch(postsActions.initPosts(useR.id));
     }
-  }, [author?.id]);
+  }, [useR?.id]);
 
   return (
     <main className="section">
-      {/* Learn the Redux Toolkit usage example in src/app and src/features/counter */}
-      <Counter />
-
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector value={author} onChange={setAuthor} />
+                <UserSelector />
               </div>
 
               <div className="block" data-cy="MainContent">
-                {!author && (
+                {!useR && (
                   <p data-cy="NoSelectedUser">
                     No user selected
                   </p>
                 )}
 
-                {author && !loaded && (
+                {useR && loading && (
                   <Loader />
                 )}
 
-                {author && loaded && hasError && (
+                {useR && !loading && error && (
                   <div
                     className="notification is-danger"
                     data-cy="PostsLoadingError"
                   >
-                    Something went wrong!
+                    {error}
                   </div>
                 )}
 
-                {author && loaded && !hasError && posts.length === 0 && (
+                {useR && !loading && !error && !posts.length && (
                   <div className="notification is-warning" data-cy="NoPostsYet">
                     No posts yet
                   </div>
                 )}
 
-                {author && loaded && !hasError && posts.length > 0 && (
-                  <PostsList
-                    posts={posts}
-                    selectedPostId={selectedPost?.id}
-                    onPostSelected={setSelectedPost}
-                  />
+                {useR && !loading && !error && !!posts.length && (
+                  <PostsList />
                 )}
               </div>
             </div>
           </div>
 
           <div
             data-cy="Sidebar"
-            className={classNames(
-              'tile',
-              'is-parent',
-              'is-8-desktop',
-              'Sidebar',
-              {
-                'Sidebar--open': selectedPost,
-              },
-            )}
+            className={posT
+              ? 'tile is-parent is-8-desktop Sidebar Sidebar--open'
+              : 'tile is-parent is-8-desktop Sidebar'}
```

Comment:
Use [classnames](https://www.npmjs.com/package/classnames) library to add classes conditionally